### ISSUE TITLE
Sageattention breaks compiling with onediff.

### ISSUE THREAD

**Issue by Ph0rk0z** [kijai/ComfyUI-CogVideoXWrapper#133]
Installed sageattention. It gave a speed boost. Even if I don't use fp8 quant, compile with onediff fails. Says cannot find a compatible kernel from triton. Uninstalling sageattention works but then it can't be used otherwise. 3090 on linux.

**Comment by kijai** [comment]
I think this is to be expected, I doubt it can work with onediff without specific support from them, I don't know enough of this to say if it's something that will be possible.

**Comment by Ph0rk0z** [comment]
I think the fix would be to not have it use sageattention when onediff compile is on, rather than making them work together. Right now it's beating both compilation methods anyways. I get down to 9s/it.

Speaking of compile, the torch.compile also fails on the 2nd gen when the seed is changed. I'd like to try compiling with it that way, but I never get to enjoy the fruits of my labor for that reason.

Have you tried their other onediff backend or is it incompatible with the model arch?

**Comment by Ph0rk0z** [comment]
I guess I will try oneflow and see if it errors out.

**Comment by kijai** [comment]
> I guess I will try oneflow and see if it errors out.
> 
> edit: played with some of the options you send to nexfort and they speed things up when not using fp8. when the first one was set false, compiling was almost slower than not.

That's curious, it makes no difference to me, but someone else told the same that it's slower... it was taken from the CogVideo example on the onediff repo, probably best to change it then.

**Comment by Ph0rk0z** [comment]
Yup, I turned on the other one too, but I don't think that makes a difference in speed. Made the videos less blurry when using the LCM scheduler.

I seem to get the best perf when setting things to FP16 but then I get a black image. Its as if there are NaN in the output. maybe it's so fast because it's multiplying 0's?

I also tried to use ondiff, but it fails on the attention function. The 2 shapes it torch.cat's at line ~63 are not compatible.

**Comment by tavyscrolls** [comment]
> Nexfort is the only one I've gotten to work, never got torch.compile to work at all.

Does this mean you also get a mosaic of pixel patches with sageattention installed? Fresh environment, all python versions none of them produce images but removing sageattention fixes it

**Comment by kijai** [comment]
> > Nexfort is the only one I've gotten to work, never got torch.compile to work at all.
> 
> Does this mean you also get a mosaic of pixel patches with sageattention installed? Fresh environment, all python versions none of them produce images but removing sageattention fixes it

I haven't tried it with onediff, otherwise it's been working fine and I don't see any quality changes.
